lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -163,9 +170,19 @@ class DiscoverLoops {
             recordDataflow(connect.getDest(), connect.getSrc());
           })
           .Case<mlir::UnrealizedConversionCastOp>([&](auto) {
-            // Assume external dialects mixed into FIRRTL do not close cycles.
+            // Casts are cast-like regardless of source
```

Comment:
`CastOpInterface` ? 

Also, unrealized conversion cast can have any number of results -- not just one. ( https://mlir.llvm.org/docs/Dialects/Builtin/#builtinunrealized_conversion_cast-unrealizedconversioncastop ).

Reply:
UnrealizedConversionCastOp doesn't implement CastOpInterface!